framework/contracts/native/account-factory/src/contract.rs

```diff
@@ -33,8 +33,12 @@ pub fn instantiate(
     cw2::set_contract_version(deps.storage, ACCOUNT_FACTORY, CONTRACT_VERSION)?;
 
     CONFIG.save(deps.storage, &config)?;
-    // Set up the admin as the creator of the contract
-    cw_ownable::initialize_owner(deps.storage, deps.api, Some(info.sender.as_str()))?;
+    // Set up the admin
+    let owner = match msg.admin.as_deref() {
```

Comment:
This code is repeated 4 times, could we have a function instead that initializes the owner base on a optional admin field ?

Reply:
True, thanks